cmd/createcluster.go

```diff
@@ -672,17 +684,39 @@ func getValidators(
 			return nil, errors.Wrap(err, "builder registration to cluster object")
 		}
 
+		var partialDepositData []cluster.DepositData
+
+		// The loop over partial amounts to collect PartialDepositData
+		for i := range depositDatas {
```

Comment:
what if `depositDatas` looks like `map[pubkey][]eth2p0.DepositData` where pubkey is the dv pubkey? Would this code segment looked more clean?

Reply:
Reworked.